testng-core/src/main/java/org/testng/JarFileUtils.java

```diff
@@ -104,15 +104,20 @@ private boolean testngXmlExistsInJar(File jarFile, List<String> classes) throws
       Collection<XmlSuite> parsedSuites = Parser.parse(suitePath, processor);
       delete(file);
       for (XmlSuite suite : parsedSuites) {
-        // If test names were specified, only run these test names
+        // If test names were specified, only run these test names. If any test names missed, then
+        // won't run any test names. (default legacy logic)
         if (testNames != null) {
           TestNamesMatcher testNamesMatcher = new TestNamesMatcher(suite, testNames);
-          testNamesMatcher.validateMissMatchedTestNames(ignoreMissedTestNames);
-          suites.addAll(testNamesMatcher.getSuitesMatchingTestNames());
+          boolean validationResult =
+              testNamesMatcher.validateMissMatchedTestNames(ignoreMissedTestNames);
+          if (validationResult) {
+            suites.addAll(testNamesMatcher.getSuitesMatchingTestNames());
+          }
```

Comment:
else print warning message to tell there is no test to run?

Reply:
if it's false, an exception will be thrown.
false means there is no any test names matching the existing ones, and nothing to run.

Reply:
false here only means the test names are not found in the current suite.

Reply:
Print error message and return false here.
However, it's hard to imagine one situation that:
1. given test names are valid: non-null, non-empty, non-all test names are null/empty/blank.
2. the missed test names are empty and matched test names are also mepty. 
According to current logic, it would be never to happen.